v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -455,15 +455,15 @@ private static PTransform<PBegin, PCollection<SourceRow>> getReadWithUniformPart
                     sourceTableSchema,
                     config.shardID()))
             .setWaitOn(config.waitOn())
-            /* The following setting limits number of stages provisioned for the split process.
-             * Currently we mostly deal with auto incrementing keys, so we don't need a split depth to make the partition uniform, unless there is a large dataset with a lot of holes.
-             * TODO(vardhanvthigle): if index is not of the type of a single auto incrementing key, don't set this.
-             */
-            .setSplitStageCountHint(0L)
             .setDbParallelizationForSplitProcess(config.dbParallelizationForSplitProcess())
             .setDbParallelizationForReads(config.dbParallelizationForReads())
             .setAdditionalOperationsOnRanges(config.additionalOperationsOnRanges());
 
+    if (config.splitStageCountHint() >= 0) {
```

Comment:
Noticed there is no code change in this PR setting split counts to log(number of partitions) anywhere. Was it already handled during mysql implementation?

Also, is this correct given when the value is 0, we disable hints?

Reply:
It's handeld in unifromPartition Logic which is common across mySql and PG.
If we don't set anything here, RWUPT automatically infers the count as log2(numPartitions).